pants-plugins/internal_plugins/versioned_setup/register.py

```diff
@@ -0,0 +1,81 @@
+import os.path
+
+from pants.backend.python.goals.setup_py import SetupKwargs, SetupKwargsRequest
+from pants.engine.fs import DigestContents, GlobMatchErrorBehavior, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+
+
+class VersionedSetupKwargsRequest(SetupKwargsRequest):
+    @classmethod
+    def is_applicable(cls, _: Target) -> bool:
+        # We always use our custom `setup()` kwargs generator for `python_distribution`
+        # targets in this repo.
+        return True
+
+
+@rule
+async def setup_kwargs_plugin(request: VersionedSetupKwargsRequest) -> SetupKwargs:
+    kwargs = request.explicit_kwargs.copy()
+
+    # Add classifiers. We preserve any that were already set.
+    standard_classifiers = [
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: MIT",
+        "Operating System :: MacOS :: MacOS X",
+        "Operating System :: POSIX :: Linux",
+        "Programming Language :: Python",
+        "Topic :: Software Development :: Build Tools",
+    ]
+    kwargs["classifiers"] = [*standard_classifiers, *kwargs.get("classifiers", [])]
+    project_urls = {
+        "Source": "https://github.com/CodeForAfrica/api",
+        "Tracker": "https://github.com/CodeForAfrica/api/issues",
+        "Twitter": "https://twitter.com/Code4Africa",
+    }
+    kwargs["project_urls"] = {**project_urls, **kwargs.get("project_urls", {})}
+    default_kwargs = dict(
```

Comment:
This is okay but maybe we can use a consistent dictionary declaration? Unless there are some other reasons.

Reply:
Not sure I fully get you but if the question is why are we merging these two dictionaries, then it's because some projects may have their own URLs e.g. DebunkBot has its own Twitter account separate from the general Code for Africa account.

Reply:
Ahh... yes @esirK, `{}` should be all we need these days.